src/systemTest/java/com/flightstats/hub/resilient/BaseTest.java

```diff
@@ -15,15 +16,15 @@
 @Slf4j
 public class BaseTest {
 
-    protected Injector injector = Guice.createInjector(new GuiceModule());
```

Comment:
I like this approach better. One has to just extend the test from this base class and  all required components for the test is already setup. The base test reads properties file, creates retrofit clients for hub, callback server and sets up Guice DI. The required dependencies in the tests can be injected using annotations. Baseclass is encapsulating all the details from the tests.
Setting up configuration in each test will be redundant.

Reply:
I think I can meet you halfway and keep this base class. Instead of `BaseTest` call it `GuiceInjection` or something similar...and then move all the protected methods out into another client-related helper class that can be injected into the tests. That way the only behavior in this class would be to handle injection. Limiting behavior plus renaming will reduce the likelihood that folks will put other "helping" behavior in here. Might make sense to remove the @Slf4j annotation, too, so we don't have to override stuff in all the tests.